src/contracts/QBridge.h

```diff
@@ -0,0 +1,362 @@
+        uint8 orderType;                     // Type of order (e.g., mint, transfer)
+        uint8 status;                        // Order status (e.g., Created, Pending, Refunded)
+        bit fromQubicToEthereum;             // Direction of transfer
+    };
+
+    // Input and Output Structs
+    struct createOrder_input {
+        id ethAddress;
+        uint64 amount;
+        bit fromQubicToEthereum;
+    };
+
+    struct createOrder_output {
+        uint8 status;
+        array<uint8, 32> message;
+    };
+
+    struct completeOrder_input {
+        uint64 orderId;
+    };
+
+    struct completeOrder_output {
+        uint8 status;
+        array<uint8, 32> message;
+    };
+
+    struct refundOrder_input {
+        uint64 orderId;
+    };
+
+    struct refundOrder_output {
+        uint8 status;
+        array<uint8, 32> message;
+    };
+
+    // Order Response Structure 
+    struct OrderResponse {
+        uint64 orderId;                      // Order ID as uint64
+        id originAccount;                    // Origin account
+        id destinationAccount;               // Destination account
+        uint64 amount;                       // Amount as uint64
+        array<uint8, 64> memo;               // Notes or metadata
+        uint32 sourceChain;                  // Source chain identifier
+    };
+
+    struct getOrder_input {
+        uint64 orderId;
+    };
+
+    struct getOrder_output {
+        uint8 status;
+        array<uint8, 32> message;
+        OrderResponse order;                 // Updated response format
+    };
+
+
+private:
+    // Contract State
+    QPI::HashMap<uint64, BridgeOrder, 256> orders; // Storage for orders (fixed size)
+    uint64 nextOrderId;                            // Counter for order IDs
+    uint64 lockedTokens;                           // Total locked tokens in the contract (balance)
+    uint64 transactionFee;                         // Fee for creating an order
+    id admin;                                      // Admin address
+    QPI::HashMap<id, bit, 16> managers;            // Managers list
+    uint64 totalReceivedTokens;                    // Total tokens received
+
+    // Internal methods for admin/manager permissions using qpi CALLS
+    PRIVATE_FUNCTION_WITH_LOCALS(isAdmin)
+        output = (qpi.invocator() == state.admin);
+    _
+
+        PRIVATE_FUNCTION_WITH_LOCALS(isManager)
+        output = state.managers.get(qpi.invocator(), bit());
+    _
+
+        /*
+        // Check if the invocator is the admin
+        bool isAdmin() const {
+            return qpi.invocator() == state.admin;
+        }
+
+        // Check if the invocator is a manager
+        bool isManager() const {
+            bit isManager = false;
+            state.managers.get(qpi.invocator(), isManager);
+            return isManager;
+        }*/
+
+public:
+    // Create a new order and lock tokens
+    PUBLIC_FUNCTION_WITH_LOCALS(createOrder)
```

Comment:
Thank you for pointing this out. I’ve updated the createOrder function as follows:

Changed PUBLIC_FUNCTION_WITH_LOCALS(createOrder) to PUBLIC_PROCEDURE_WITH_LOCALS(createOrder) since it modifies the state. And it is defined the required createOrder_locals struct. Then, I replaced 'newOrder' with locals.newOrder throughout the function.

Hope that with this modifications it ensures compliance with the restriction on using the regular stack. If there are further updates to the documentation regarding this, I’ll make sure to adjust accordingly.